test/integration/sanity/file-uploads.test.js

```diff
@@ -1,76 +1,247 @@
-describe('File uploads', function() {
```

Comment:
confine the sanity test to be a single happy flow and move rest out.

+ add test for all sad cases.
+ add test for cases where resolver is not sent or resolver is partially defined